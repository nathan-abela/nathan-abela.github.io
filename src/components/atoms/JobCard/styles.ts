import styled from 'styled-components';

const StyledJobCard = styled.div`
	margin-bottom: 40px;
`;

const Flex = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	margin-bottom: 8px;

	@media screen and (min-width: 768px) {
		flex-direction: row;
	}
`;

export { Flex, StyledJobCard };
