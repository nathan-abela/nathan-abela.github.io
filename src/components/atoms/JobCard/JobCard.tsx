import { FC } from 'react';
import { format } from 'date-fns';

import { IJob } from '@Types';
import { Flex, StyledJobCard } from './styles';

export interface PositionProps {
	job: IJob;
}

const JobCard: FC<PositionProps> = ({ job }) => {
	return (
		<StyledJobCard>
			<Flex>
				<div>
					<strong>{job.jobTitle}</strong> -{' '}
					{job.website ? (
						// eslint-disable-next-line prettier/prettier
						<a
							href={job.website}
							target="_blank"
							rel="noopener noreferrer"
						>
							{job.company}
						</a>
					) : (
						job.company
					)}
				</div>

				<div>
					{format(new Date(job.startDate), 'MMMM yyyy')} -{' '}
					{job.endDate ? format(new Date(job.endDate), 'MMMM yyyy') : 'Present'}
				</div>
			</Flex>

			<small>
				<em>{job.location}</em>
			</small>

			<p>{job.description}</p>

			{job.bullets.length > 0 && (
				<ul style={{ paddingLeft: 16 }}>
					{job.bullets.map((bullet, j) => (
						<li key={j}> &bull; {bullet} </li>
					))}
				</ul>
			)}

			<small>
				<em>
					<strong>Technologies:</strong> {job.technologies.join(', ')}
				</em>
			</small>
		</StyledJobCard>
	);
};

export { JobCard };
