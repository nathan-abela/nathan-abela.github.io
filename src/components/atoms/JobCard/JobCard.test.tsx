import { render } from '@testing-library/react';

import { JobCard } from './JobCard';

describe('Job Card', () => {
	it('renders correctly', () => {
		const { container } = render(
			<JobCard
				job={{
					company: 'Acme Corporation',
					website: 'https://wikipedia.org/wiki/Acme_Corporation',
					location: 'Valletta, Malta',
					jobTitle: 'Software Engineer',
					startDate: 'January 2020',
					endDate: 'January 2022',
					description: 'Worked on a variety of software projects.',
					bullets: [
						'Developed and maintained web applications using Angular.',
						'Collaborated with cross-functional teams to deliver high-quality software.',
						'Participated in code reviews and provided constructive feedback.',
					],
					technologies: ['JavaScript', 'TypeScript', 'Node.js', 'Angular', 'HTML', 'CSS'],
				}}
			></JobCard>
		);
		expect(container).toMatchSnapshot();
	});
});
