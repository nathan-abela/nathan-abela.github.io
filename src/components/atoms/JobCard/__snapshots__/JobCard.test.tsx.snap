// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Card renders job card correctly 1`] = `
<div>
  <div
    class="styles__StyledJobCard-sc-1msshzv-0 kONyWZ"
  >
    <div
      class="styles__Flex-sc-1msshzv-1 hUDaJt"
    >
      <div>
        <strong>
          Software Engineer
        </strong>
         -
         
        <a
          href="https://wikipedia.org/wiki/Acme_Corporation"
          rel="noopener noreferrer"
          target="_blank"
        >
          Acme Corporation
        </a>
      </div>
      <div>
        January 2020
         -
         
        January 2022
      </div>
    </div>
    <small>
      <em>
        Valletta, Malta
      </em>
    </small>
    <p>
      Worked on a variety of software projects.
    </p>
    <ul
      style="padding-left: 16px;"
    >
      <li>
         • 
        Developed and maintained web applications using Angular.
         
      </li>
      <li>
         • 
        Collaborated with cross-functional teams to deliver high-quality software.
         
      </li>
      <li>
         • 
        Participated in code reviews and provided constructive feedback.
         
      </li>
    </ul>
    <small>
      <em>
        <strong>
          Technologies:
        </strong>
         
        JavaScript, TypeScript, Node.js, Angular, HTML, CSS
      </em>
    </small>
  </div>
</div>
`;
